File: Binary Search/Python/H-Index II.py
// TC:O(log n) SC:O(1)
class Solution:
    def hIndex(self, citations: List[int]) -> int:
     n=len(citations)
     low=0
     high=n-1
     while low<=high:
        mid=low+(high-low)//2
        h=n-mid
        if citations[mid]==h:return h
        if citations[mid]<h:low=mid+1
        else:high=mid-1
     return n-low
        

File: Binary Search/H-Index II.cpp
// TC:O(log n) SC:O(1)
class Solution {
public:
    int hIndex(vector<int>& citations) {
            int n=citations.size();
            int low=0,high=n-1;
            while(low<=high){
                int mid=low+(high-low)/2;
                int h=n-mid;
                if(citations[mid] == h) return h;
                if(citations[mid]<h) low=mid+1;
                else high=mid-1;
            }
            return n-low;
    }
};
